### DISCUSSION TITLE
ArcGIS Pro INSPIRE_MD2 Add-In

### DISCUSSION THREAD

**Discussion by ffarr2019** [INSPIRE-MIF/helpdesk-validator#807] [General]
I exported an XML metadata (v2) file using the ArcGIS Pro module "INSPIRE_MD2 Add-In".  the exported metadata is not in GMD element, however, it still passed the validator.  Is the XML still considered as INSPIRE compliant and acceptable?

_sample of exported XML_

> <?xml version="1.0" encoding="UTF-8"?>
> <MD_Metadata xmlns="http://www.isotc211.org/2005/gmd" xmlns:gco="http://www.isotc211.org/2005/gco" xmlns:gts="http://www.isotc211.org/2005/gts" xmlns:srv="http://www.isotc211.org/2005/srv" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:gmx="http://www.isotc211.org/2005/gmx" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
> <fileIdentifier>
> <gco:CharacterString>4504DB76-FF5E-4F56-9052-503F6C6F1E44</gco:CharacterString>
> </fileIdentifier>
> <language>
> <LanguageCode codeList="http://www.loc.gov/standards/iso639-2" codeListValue="eng" codeSpace="ISO639-2">eng</LanguageCode>
> </language>
> <characterSet>
> <MD_CharacterSetCode codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#MD_CharacterSetCode" codeListValue="utf8" codeSpace="ISOTC211/19115">utf8</MD_CharacterSetCode>
> </characterSet>
> <hierarchyLevel>
> <MD_ScopeCode codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#MD_ScopeCode" codeListValue="dataset" codeSpace="ISOTC211/19115">dataset</MD_ScopeCode>
> </hierarchyLevel>
> <hierarchyLevelName>
> <gco:CharacterString>dataset</gco:CharacterString>
> </hierarchyLevelName>
> <contact>
> <CI_ResponsibleParty>
> <individualName>

![image](https://user-images.githubusercontent.com/32981299/126261850-42603193-42c8-400f-b58b-ad849f585049.png)

**Comment by fabiovin** [comment]
Dear @ffarr2019,

thank you for raising the case, we never considered this type of encoding.

In any case, an XML without `gmd:`namespace declared for each element is valid, at least from a technical point of view.

We will discuss internally the case and report the result.

At the moment, you are INSPIRE compliant! ;-)

**Comment by ffarr2019** [comment]
Thanks a lot Fabio :)

Frank

**Comment by jsaligoe** [comment]
Hi Frank, good question – a bit of follow-up as this relates to use of default namespaces. A default namespace declaration applies to all unprefixed element names within its scope and can create well-formed XML. I illustrate it below with an example screenshot of an ISO 19139 XML document.

Different XML schemas can define XML elements with the same name but with different intentions regarding the content of the element. Elements derived from different XML schemas can be disambiguated by specifying a namespace.
 
In this example, two namespaces are highlighted at the top of the XML file in the screenshot, one for the GMD namespace (orange) and one for the GCO namespace (blue). Looking at the blue GCO namespace first, you can see in the way it is declared "xmlns:gco" that the full schema URI is associated with the prefix GCO. The CharacterString element is defined in the schema associated with the GCO namespace, and so when it is specified it is with the namespace prefix, i.e. gco:CharacterString.

![default-namespace](https://user-images.githubusercontent.com/49758885/128427804-88b924e9-659d-4bb6-8b6d-852192d23a22.jpg)

Looking at the orange GMD namespace declaration, you can see that no element prefix is defined. That means every XML element in the document that does not have a prefix is part of the GMD namespace. This is known as a default namespace. With ISO 19139 XML documents, it is particularly appropriate to use a default namespace since almost every XML element in the document is derived from the same GMD namespace. All of the XML elements identified in orange do not have a specified namespace prefix. However, by virtue of the default namespace declaration at the top of the file, they are defined as belonging to the GMD namespace.
 
XML documents that make use of a default namespace are entirely valid. The use of default namespaces is defined in the W3C XML specification. You can learn more about XML namespaces in general, and default namespaces in particular from the following sources:
* https://www.w3.org/TR/xml-names/#defaulting
* https://www.oracle.com/technical-resources/articles/srivastava-namespaces.html
* https://docs.microsoft.com/en-us/dotnet/standard/data/xml/managing-namespaces-in-an-xml-document

@Fabio, thanks also for the confirmation of compliance ;) We have recently updated ArcGIS Pro, ArcGIS Online, and ArcGIS Enterprise to support the latest INSPIRE Metadata v2 to ensure our users can create metadata that do “Pass” validation. Because knowing how to complete INSPIRE metadata can also be a challenge, we’ve also created some step-by-step tutorials to help guide users through the process. https://www.esri.com/arcgis-blog/products/arcgis-online/sharing-collaboration/inspire-metadata-open-data-in-arcgis/   BTW, Nice job on the new Validator release - it make the process so much easier!

Best regards, Jill
